internal/predicates/osbuildconfig_test.go

```diff
@@ -49,6 +52,23 @@ var _ = Describe("OSBuildConfig reconciliation predicate", func() {
 				},
 			},
 		),
+		Entry("when template version changed and template trigger is explicitly enabled",
```

Comment:
Should we also check when the template trigger is not set (default behavior)?

Reply:
It's covered here: https://github.com/project-flotta/osbuild-operator/pull/63/files#diff-4d944c176d5a7c6d7e5a0e1c0bd2ba6bd49d8e0479a6d8f225840691d81b8ffaR35